frontend/src/pages/Settings.tsx

```diff
@@ -20,83 +20,70 @@ import {
 
-type DataSource = {
-  id: string;
-  name: string;
-  connected: boolean;
-  letter: string;
-  bgClass: string;
-  textClass: string;
-};
+function BrandIcon({ brandId, size = 40 }: { brandId: string; size?: number }) {
+  const [src, setSrc] = useState(`/static/assets/logos/${brandId}.svg`);
 
-const DATA_SOURCES = [
-  {
-    id: "amazon",
-    name: "Amazon",
-    connected: true,
-    letter: "A",
-    bgClass: "bg-orange-500",
-    textClass: "text-white",
-  },
-  {
-    id: "goodreads",
-    name: "Goodreads",
-    connected: true,
-    letter: "G",
-    bgClass: "bg-amber-500",
-    textClass: "text-white",
-  },
-  {
-    id: "doordash",
-    name: "DoorDash",
-    connected: false,
-    letter: "D",
-    bgClass: "bg-purple-500",
-    textClass: "text-white",
-  },
-  {
-    id: "bbc",
-    name: "BBC",
-    connected: true,
-    letter: "B",
-    bgClass: "bg-neutral-700",
-    textClass: "text-white",
-  },
-  {
-    id: "cnn",
-    name: "CNN",
-    connected: false,
-    letter: "C",
-    bgClass: "bg-red-500",
-    textClass: "text-white",
-  },
-  {
-    id: "zillow",
-    name: "Zillow",
-    connected: true,
-    letter: "Z",
-    bgClass: "bg-blue-800",
-    textClass: "text-white",
-  },
-];
+  function handleError() {
+    if (src.endsWith(".svg")) {
+      setSrc(`/static/assets/logos/${brandId}.png`);
+      return;
+    }
+    if (src.endsWith(".png")) {
+      setSrc("/static/assets/logos/default.svg");
+    }
+  }
+
+  return (
+    <img
+      src={src}
+      alt={brandId}
+      width={size}
+      height={size}
+      loading="lazy"
+      onError={handleError}
+      className="h-10 w-10 rounded-xl bg-gray-50 object-contain p-1"
+    />
+  );
+}
 
 export default function Settings() {
   const [isRecordingEnabled, setIsRecordingEnabled] = useState(false);
   const [recordingDelay, setRecordingDelay] = useState(5);
 
-  // TODO: get this from API
-  const [dataSources, setDataSources] = useState<DataSource[]>(DATA_SOURCES);
+  const [brands, setBrands] = useState<BrandState[]>([]);
 
-  function toggleDataSource(id: string) {
-    // TODO: integrate to API
-    setDataSources((prev) =>
-      prev.map((ds) =>
-        ds.id === id ? { ...ds, connected: !ds.connected } : ds,
+  function toggleBrandEnabled(brand_id: string) {
+    const brand = brands.find((b) => b.brand_id === brand_id);
+    if (!brand) {
+      return;
+    }
+
+    setBrands(
+      brands.map((b) =>
+        b.brand_id === brand_id ? { ...b, enabled: !b.enabled } : b,
       ),
     );
+    apiService
+      .updateBrandEnabled(brand_id, !brand.enabled)
+      .then(() => {})
+      .catch((error) => {
+        console.error(error);
+        setBrands(
+          brands.map((b) =>
+            b.brand_id === brand_id ? { ...b, enabled: !b.enabled } : b,
+          ),
+        );
+      });
   }
 
+  useEffect(() => {
+    apiService.fetchBrands().then((result) => {
+      setBrands(result);
+    });
+  }, []);
+
```

Comment:
feels like we need to put this logic into its own hooks `useBrands` -> better if we can just fetch it once and store it globally.